sound/soc/sof/pcm.c

```diff
@@ -71,6 +71,11 @@ void snd_sof_pcm_period_elapsed_work(struct work_struct *work)
 	snd_pcm_period_elapsed(sps->substream);
 }
 
+void snd_sof_pcm_init_elapsed_work(struct work_struct *work)
```

Comment:
This splitting into an init function and creation of a work_struct feels like it could be done in a separate commit.

Reply:
feels like it but in this case you want to see the general picture. I think it is better this way.